genie-agent/src/main/java/com/netflix/genie/agent/cli/ArgumentValidators.java

```diff
@@ -53,6 +54,25 @@ public void validate(final String name, final String value) throws ParameterExce
         }
     }
 
+    /**
+     * Validates a string parameter is a valid S3 uri.
+     */
+    public static class S3URIValidator implements IParameterValidator {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void validate(final String name, final String value) throws ParameterException {
```

Comment:
There is some code we already have for this in `S3FileTransferImpl` that we may want to refactor for reuse. It doesn't have to be done as part of this PR but a `tech-debt` ticket can be filed. That code is a little more robust than this validation:

```    @VisibleForTesting
    AmazonS3URI getS3Uri(final String path) throws GenieBadRequestException {
        if (!S3_PREFIX_PATTERN.matcher(path).matches()) {
            throw new GenieBadRequestException(String.format("Invalid prefix in path for s3 file %s", path));
        }
        // Delegate validation and parsing to AmazonS3URI.
        // However it cannot handle "s3n://", so strip the 'n'
        final String adjustedPath = path.replaceFirst("^s3n://", "s3://");
        final AmazonS3URI uri;
        try {
            uri = new AmazonS3URI(adjustedPath, false);
        } catch (IllegalArgumentException e) {
            throw new GenieBadRequestException(String.format("Invalid path for s3 file %s", path), e);
        }
        if (StringUtils.isBlank(uri.getBucket()) || StringUtils.isBlank(uri.getKey())) {
            throw new GenieBadRequestException(String.format("Invalid blank components in path for s3 file %s", path));
        }

        final boolean bucketPassesStrictValidation = S3_BUCKET_PATTERN.matcher(uri.getBucket()).matches();
        final boolean keyPassesStrictValidation = S3_KEY_PATTERN.matcher(uri.getKey()).matches();
        // URL fails strict validation check!
        if (!bucketPassesStrictValidation || !keyPassesStrictValidation) {
            if (this.s3FileTransferProperties.isStrictUrlCheckEnabled()) {
                throw new GenieBadRequestException(
                    String.format(
                        "Invalid bucket %s in path for s3 file %s",
                        uri.getBucket(),
                        path
                    )
                );
            } else {
                log.warn("S3 URL fails strict validation: \"{}\"", path);
                this.urlFailingStrictValidationCounter.increment();
            }
        }
        return uri;
    }```

Reply:
Will do

Reply:
Done